cms/djangoapps/contentstore/views/item.py

```diff
@@ -103,6 +103,28 @@ def _filter_entrance_exam_grader(graders):
     return graders
 
 
+def set_discussion_toggle(value, xblock, user):
+    """
+    Recursively update all children of given xblock in modulestore
+    """
+    is_block_with_discussion_enabled = (
+        xblock.location.block_type in ["vertical", "sequential", "chapter", "course"]
+    )
+    if not all([isinstance(value, bool), is_block_with_discussion_enabled]):
+        return
```

Comment:
I'm not sure why you're doing this and why it's being done is such an odd manner!
We're generally not testing the types of parameters so why the need to check if value is a `bool`?
There are just two items here, so why not just use an `and` instead of putting them in an array and calling all?
```suggestion
    if not xblock.location.block_type in ["vertical", "sequential", "chapter", "course"]:
        return
```

Reply:
@xitij2000 If we don't check for `bool` wouldn't it violate the `discussion_enabled` field's type which is `bool`.

Reply:
> @xitij2000 If we don't check for `bool` wouldn't it violate the `discussion_enabled` field's type which is `bool`?

To me that only makes sense if:
- The value comes from an untrusted place where it could have any value. (Not the case here since it's an internal API). 
- A value of the wrong type will cause an error. (Not the case here). 

We are after all not validating that xblock is actually an XBlock, of that user is actually a user object. 
In this case it is important that we NOT check the type. What if it is a string like "true"? This code will reject it and not work, however that is a perfectly valid way to set this field to True. 

XBlocks [already have a machinery in place](https://github.com/edx/XBlock/blob/79db8bde016e38fa21d3ad985aa3da6e7cdf5117/xblock/fields.py#L737-L756) to automatically convert this to a valid representation.